src/angular/src/app/pages/logs/logs-page.component.scss

```
@@ -91,15 +104,63 @@
         }
+    }
+
+    /* Live-log rows are virtualized with a FIXED itemSize (#539), so each row
+       must be exactly --log-row-height tall regardless of message length or a
+       traceback. The message line is kept single-line (ellipsis + title tooltip)
+       and the traceback opens in an absolutely-positioned popover that does NOT
+       grow the row, preserving the fixed-size scroll math. Keep this height in
+       sync with LogsPageComponent.ROW_HEIGHT. */
+    .log-viewport p.record {
+        --log-row-height: 22px;
+        position: relative;
```

Stylelint (17.12.0)

[error] Expected empty line before declaration (declaration-empty-line-before)

(declaration-empty-line-before)